subsys/shell/Kconfig.backends

```diff
@@ -24,14 +24,13 @@ config SHELL_BACKEND_SERIAL
 
 if SHELL_BACKEND_SERIAL
 
-if !HAS_DTS
```

Comment:
why you removed that? It is (was) needed for boards without DTS

Reply:
The functionality is the same, its just now done consistently regardless of DTS.  DTS will still provide  this information, and the `default "$(dt_str_val,DT_UART_SHELL_ON_DEV_NAME)" if HAS_DTS` is how that's happening.